### ISSUE TITLE
need to change 1 line to fix a deprecation error that is printing alot and makes debug harder

### ISSUE THREAD

**Issue by sinamcr7** [usnistgov/OpenSeadragonFiltering#29]
the error says :
[tile-loaded] event 'image' has been deprecated. Use 'data' property instead.

and need to change 'event.image' to 'event.data' in code to fix it

**Comment by iangilman** [comment]
For context, this deprecation is due to https://openseadragon.github.io/examples/advanced-data-model/. FYI, @Aiosa.

**Comment by Aiosa** [comment]
I am not aware of anything in the viewer that should cause a depreciation warn, probably a custom code or a plugin. It is deprecated. You need to change your code to use `data` property since future versions might drop its support.

**Comment by iangilman** [comment]
@Aiosa This is the GitHub repo for OpenSeadragonFiltering. The warning comes whenever you use this plugin. I don't know that there's anything to be done about it... Of course, ideally the plugin would get updated, but I don't think it's being maintained that actively these days. Anyway, just noticing that this is a side effect of the deprecation warning; it affects existing plugins.

**Comment by Aiosa** [comment]
Ah, thanks for the heads up 😄 and I guess that's just natural for a software without development for 5 years or so.. Maybe it's time to create a fork and deprecate the old lib in OSD docs...

**Comment by iangilman** [comment]
Indeed! This is a popular one! We just need someone who's interested in taking it over.

**Comment by Aiosa** [comment]
Hmm actually I will wait until I look at the WebGL renderer we plan, it could drastically improve the overall usability if we could create a new image processing plugin over it, and it would nicely test the extensibility of the renderer.

**Comment by iangilman** [comment]
Agreed!

**Comment by abiswas97** [comment]
@Aiosa @iangilman I stumbled across this issue while looking into using this plugin. Is there an open repo that's currently working on a new filtering plugin?

On a side note, I noticed that the webgl renderer is now available on main - is the plan to roll out the webgl renderer with the next major release of OSD?

**Comment by Aiosa** [comment]
Not officially, though I am 'working' on the new filtering plugin that adheres to the proposed cache overhaul for osd. Working is a bit strong, I could not find time for it for several weeks already. But I hope I am going to finish it. From June on I could be a bit less busy.

Regarding the release, we are
- a bit waiting for people to try out the drawer and report issues, so that the release is at least a bit stable
- maybe (not sure) waiting for the cache overhaul, which fixes the data access pipeline (mentioned above, e.g. webgl drawer needed additional events to be functional, the new cache system would handle it uniformly), but it is very tricky to implement correctly as it uses async pipeline and it is also the backbone of OSD data flow, and it also provides the data interface for users so it must be simple enough

**Comment by abiswas97** [comment]
Are you open to contributions on the filtering plugin front? I'm only just starting to work with image data manipulation, but would love to contribute where possible. If yes, is [this repo](https://github.com/Aiosa/OpenSeadragonFiltering) the one I should be looking at?

**Comment by Aiosa** [comment]
Yes, the linked repository is the implementation of the filtering plugin for the cache overhaul. However, what is not finished is the cache system not the plugin.

If you want to fix the error 'image' deprecated, simply rename the property to 'data'. It was renamed since tilesource can return anything, not just image data (vectors,  zip files, text...), though usually it contains raster data.

If you want to help with the development, it's the cache overhaul that needs contribution/testing, but it is a bit more complex. We need to finish design of the data access, and test the performance/bugs of the async pipeline (possibly removing async support if we did not complete it to our liking).

**Comment by abiswas97** [comment]
Sure thing, I can help out with smaller changes while I take a closer look at the full cache overhaul and what's going on internally
